package/libnmea2000/0001-compile-whole-lib.patch

```diff
@@ -0,0 +1,29 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index c62ad3a..ee99503 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -30,8 +30,5 @@ add_compile_options(
+   -g
+ )
+ 
+-enable_testing()
+-
+ add_subdirectory(src)
+ add_subdirectory(third-party/catch)
+-add_subdirectory(test)
+diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
+index d7fab27..e61e808 100644
+--- a/src/CMakeLists.txt
++++ b/src/CMakeLists.txt
+@@ -27,6 +27,11 @@ add_library(nmea2000
+   N2kStream.cpp
+   N2kMessages.cpp
+   Seasmart.cpp
++  ActisenseReader.cpp
```

Comment:
Why are these excluded from the normal build?

Reply:
The library is originally written for Arduino. I am not overly familiar with the Arduino build process, but my guess is that Arduino builds it into the binary "by default", statically, and not into a shared library, which is what we might want to use.

I'm relatively certain that we won't need all that code in a shared lib (~100K), so I'll see what can be removed.

For now, this works and the patch is small.

swift-nav/firmware_team_planning/issues/450